src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -873,3 +873,201 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
+ * If the option is not set, or if an error is returned, @out_value will be set
+ * to %NULL.
+ *
+ * Returns: TRUE if static data superblock has signatures, FALSE otherwise.
+ *
+ * Since: 2019.6
+ */
+gboolean
+ostree_repo_static_delta_is_signed (OstreeRepo       *self,
+                                    const char       *delta_id,
+                                    char           ***out_value,
+                                    GError          **error)
+{
+  g_autoptr(GVariant) delta_meta = NULL;
+  glnx_autofd int superblock_fd = -1;
+  GVariantIter iter;
+  GVariant *item;
+  GStrv signatures = NULL;
+  int signatures_index = 0;
+  gboolean ret = FALSE;
+
+  if (out_value)
+    *out_value = NULL;
+
+  if (strchr (delta_id, '/'))
+    {
+      if (!glnx_openat_rdonly (AT_FDCWD, delta_id, TRUE, &superblock_fd, error))
+        return FALSE;
+    }
+  else
+    {
+      g_autofree char *from = NULL;
+      g_autofree char *to = NULL;
+      if (!_ostree_parse_delta_name (delta_id, &from, &to, error))
+        return FALSE;
+
+      g_autofree char *superblock_path = _ostree_get_relative_static_delta_superblock_path (from, to);
+      if (!glnx_openat_rdonly (self->repo_dir_fd, superblock_path, TRUE, &superblock_fd, error))
+        return FALSE;
+    }
+
+  g_autoptr(GVariant) delta_superblock = NULL;
+  if (!ot_variant_read_fd (superblock_fd, 0,
+                           (GVariantType*)OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT,
+                           TRUE, &delta_superblock, error))
+    return FALSE;
+
+  delta_meta = g_variant_get_child_value (delta_superblock, 0);
+  if (delta_meta == NULL)
+    {
+      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "no metadata in static-delta superblock");
+      return FALSE;
+    }
+
+  if (out_value)
+    signatures = g_malloc0_n(signatures_index + 1, sizeof(char*));
+
+  /* Check if there are signatures in the superblock */
+  g_variant_iter_init (&iter, delta_meta);
+  while ((item = g_variant_iter_next_value (&iter)))
+    {
+      g_autoptr(GVariant) key_v = g_variant_get_child_value (item, 0);
+      const char *str = g_variant_get_string (key_v, NULL);
+      if (g_str_has_prefix (str, "ostree.sign."))
+        {
+          ret = TRUE;
+          if (signatures)
+            {
+              signatures = g_realloc_n(signatures, signatures_index + 2, sizeof(char*));
+              signatures[signatures_index] = g_strdup(str + strlen("ostree.sign."));
+              signatures_index++;
+            }
+        }
+      g_variant_unref (item);
+    }
+  if (signatures)
+    signatures[signatures_index] = NULL;
+
+  if (out_value && ret)
+    *out_value = signatures;
+
+  return ret;
+}
+
+/**
+ * ostree_repo_static_delta_verify_signature:
+ * @self: Repo
+ * @delta_id: delta path
+ * @sign: Signature engine used to check superblock
+ * @error: Error
+ *
+ * Verify the superblock of static-data, except inline commits,
```

Comment:
Verify pseudo-superblock constructed from original superblock from delta file.
Does not include inlined commits or signatures themselves into signed data.